packages/mesos/extra/mesos-agent.nssm.j2

```diff
@@ -0,0 +1,6 @@
+[service]
+Description = Mesos Agent
+serviceName = mesos-agent
+Application = {{ path_to_package_dir }}\mesos-agent.exe
+AppDirectory = {{ path_to_package_dir }}
+AppParameters = --master="zk://{{ master_ip }}:2181/mesos" --launcher_dir="{{ path_to_package_dir }}\mesos-binaries" --log_dir="{{ path_to_log_dir }}" --appc_store_dir="{{ path_to_dcos_dir }}\images" --work_dir="{{ path_to_dcos_dir }}work" --runtime_dir="{{ path_to_dcos_dir }}\work" --isolation="windows/cpu,windows/mem,filesystem/windows" --containerizers="mesos,docker" --attributes="os:windows" --ip={{ local_ip }} --hostname={{ local_ip }} --resources="ports:[10000-34000]" --executor_registration_timeout=10mins --fetcher_stall_timeout=10mins
```

Comment:
Can this be created in `dcos-config-windows.yaml` so that the master IP addresses can be set there?

Reply:
We need *.nssm file for starting any service by nssm and we need to provide the correct values for starting the service by setting them in *.nssm file. At the same time dcos-config-windows.yaml is just a template that will be passed on agent node from bootstrap node and does not supposed to contain the values of the variables. 
So we need to fill up both of them the mesos.mesos-agent.nssm.j2 and the dcos-config-windows.yaml. It doesn't make sense to fill up dcos-config-windows.yaml first and then get the values from this file to mesos.mesos-agent.nssm.j2, does it?

Reply:
The `dcos-config-windows.yaml` file contains all the files that need template replacements. Parsing it should create multiple files in the filesystem, with placekeepers replaced with actual values. So, `dcos-config-windows.yaml` could contain:
```yaml
package:
  - path: {{ path_to_package_dir }}\mesos-agent.nssm
    content: |
      [service]
      Description = Mesos Agent
      serviceName = mesos-agent
      Application = {{ path_to_package_dir }}\mesos-agent.exe
      AppDirectory = {{ path_to_package_dir }}
      AppParameters = --master="zk://{{ master_ip }}:2181/mesos" --launcher_dir="{{ path_to_package_dir }}\mesos-binaries"
```
and this will create the NSSM file, with no further actions

Reply:
Alternatively, can the `{{ master_ip }}` placekeeper be replaced with `master.mesos` or `leader.mesos`? Isn't it true that at least `leader.mesos` must be set?

Reply:
Some changes have been made to the configuration files (fix b3621e648f364baf02f1246dd430674cbf2852ef). Placeholder names have been changed and agreed with @anatolii-borysov. I think he will better explain the accord of placeholder names.